pkg/csi/service/wcp/controller.go

```diff
@@ -656,7 +685,22 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 
 	// Calculate accessible topology for the provisioned volume in case of topology aware environment.
 	if isTKGSHAEnabled {
-		if zoneLabelPresent && !hostnameLabelPresent {
+		if hostnameLabelPresent {
+			// Configure the volumeTopology in the response so that the external
+			// provisioner will properly sets up the nodeAffinity for this volume.
+			if isVdppOnStretchedSVEnabled {
+				resp.Volume.AccessibleTopology = topologyRequirement.GetPreferred()
+			} else {
+				for _, hostName := range accessibleNodes {
```

Comment:
Do we have more than one accessibleNodes ever? I think we are always deploying volume on host local datastore.

Reply:
Ideally we shouldn't but I didn't want to refactor the old code as part of this MR as it's already a large one.